src/zookeeperLock.ts

```diff
@@ -81,49 +81,48 @@ export class ZookeeperLock extends EventEmitter {
         debuglog(JSON.stringify(this.config));
     }
 
+
     /**
      * create a zookeeper client which powers the lock, done when creating a new lock
      * or zk connection expires
      * @returns {Promise<any>}
      */
-    private createClient() : Promise<any> {
+    private createClient() : Promise<any>|any {
```

Comment:
the compiler doesn't seem to like the branch, likely because locators is a promise + event emitter. the `any` is to make it build, the `Promise<any>` is there to make it more obvious to the maintainer that this is returning a promise... 

I didn't put much time into this, can keep poking at it..